ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/ActionManager.java

```diff
@@ -61,7 +65,43 @@ public int getNumberOfInflightTransactions() {
     private ActionManager() {
     }
 
-    private static final ActionManager _theManager = new ActionManager();
+    /**
+     * Workaround to put BasicAction into BasicAction._allActions,
+     * which is a private static field.
+     * @param txn is the BasicAction to put back into _allActions
+     */
```

Comment:
Why not a function call from ActionManager to BasicAction, they both reside in the same package.

Reply:
Indeed. I think it makes sense to make `_allActions` private once it gets moved to `BasicAction`. Also, as we're now maintaining `_allActions` in `BasicAction`, I don't see much use for `put` and `remove` in `ActionManager`: that's why I deprecated them. Nevertheless, to keep the functionality there until we remove those two methods from the API of `ActionManager`, I had to use reflection.

Reply:
BTW, the new methods are already created with the `@Deprecated` annotation

Reply:
And since BasicAction will be maintaining the map just make put, get and remove null ops.

Reply:
You are allowed to add a new method to BasicAction you know ;-)

Reply:
I made `put` and `remove` null operations (and I deprecated them). Please, let me know if the new commit reflects your previous analysis. Thanks

Reply:
I pushed another commit to make `_allActions` visible within the package (also ActionManager is located there). I've removed `@Deprecated` from `put` and `remove`. Basically, I just moved the map from ActionManager to BasicAction. I hope this last commit reflects your previous analysis, @mmusgrov. Thanks

Reply:
Forgot to rollback `ActionTestClient`, sorry